src/local-history/local-history-manager.ts

```diff
@@ -145,14 +145,24 @@ export class LocalHistoryManager {
             // Copy the content of the current active editor.
             const fileBuffer = fs.readFileSync(document.fileName);
             const historyFilePath = path.join(workspaceFolderPath, '.local-history', historyFileName);
-            fs.writeFileSync(historyFilePath, fileBuffer);
-
             const data: HistoryFileProperties = {
                 fileName: historyFileName,
                 timestamp: timestampForProperty,
                 uri: historyFilePath,
                 parentFileName: fileFullPath.base
             };
+            const activeEditorHistories = this.historyFiles.filter(history => {
```

Comment:
@Anasshahidd21 we can optimize this to not need to iterate over all items in the history, basically to return the first value which satisfies our requirements:

```ts
const latestEditorHistory = this.historyFiles.find((history: HistoryFileProperties) => ...)
```